### PR TITLE
hotfix: cof encoder: coding weapon class

### PR DESCRIPTION
**Pull request by gucio321**

### CODE REVIEW

File: d2common/d2fileformats/d2cof/cof.go [hunk 1]
```diff
@@ -193,7 +192,15 @@ func (c *COF) Marshal() []byte {
...
-		sw.PushBytes(c.CofLayers[i].weaponClassByte...)
+		s := c.CofLayers[i].WeaponClass.String()
+
+		for j := 0; j < 4; j++ {
+			if j < len(s) {
+				sw.PushBytes(s[j])
+			} else {
+				sw.PushBytes(0)
+			}
+		}
```
Comment (gravestench) [inline d2common/d2fileformats/d2cof/cof.go:203]: what is this magic 4?

Comment (gravestench) [reply d2common/d2fileformats/d2cof/cof.go:203]: the single-letter variables deserve descriptive names

Comment (gucio321) [reply d2common/d2fileformats/d2cof/cof.go]: right, fixed in 1010353

Comment (gravestench) [reply d2common/d2fileformats/d2cof/cof.go]: ?
```golang
	const (
		maxCodeLength = 3 // we assume item codes to look like 'hax' or 'kit'
		terminator    = 0
	)

	weaponCode := c.CofLayers[i].WeaponClass.String()

	for idx, letter := range weaponCode {
		if idx > maxCodeLength {
			break
		}

		sw.PushBytes(byte(letter))
	}

	sw.PushBytes(terminator)
	```

**Review by gravestench — DISMISSED** [review]

**Review by gravestench — APPROVED** [review]

**Pull request merged**